### PR TITLE
Block tableoid access on distributed hypertable

### PR DESCRIPTION
**Pull request by pmwkaa**
This change adds a check which raises an error when system column (like
tableoid) being requested in case if data node scan is enabled.

Fixes: #2528

### CODE REVIEW

**Review by k-rus — DISMISSED** [review]
LGTM
I don't understand why it is necessary to create new distributed hypertable instead of utilising the one existing in the test.

**Review by k-rus — COMMENTED** [review]
I dismissed my review, since the PR breaks tests on PG11, where a system column OID is returned.

File: tsl/src/fdw/data_node_scan_exec.c
Comment (erimatnor) [inline tsl/src/fdw/data_node_scan_exec.c:80]: Main error should be short according to error style guide. I suggest splitting this up:

```suggestion
		ereport(ERROR,
				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
				 errmsg("system columns are not accessible on distributed hypertables"),
				 errhint("Set timescaledb.enable_per_data_node_queries=false to query system columns.")));
```

**Review by erimatnor — APPROVED** [review]
Approving, but I suggest fixing two things before merging:

* Split up error message so that main error is shorter
* Commit message should use "Fixes" or "Closes" to reference the issue as "Issue" is not a recognizable key word. https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords

**Comment by pmwkaa** [comment]
>     * Commit message should use "Fixes" or "Closes" to reference the issue as "Issue" is not a recognizable key word. https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords

That's not entirely true, github recognizes issue references in format `#number`. I was aware about the tag, I just wanted to reference the issue without doing any action to it. Anyway, updated to the 'fixes' as suggested.

File: tsl/src/fdw/data_node_scan_exec.c
Comment (k-rus) [reply tsl/src/fdw/data_node_scan_exec.c]: > "system columns are not accessible on distributed hypertables"

This suggested text is not fully correct, since system columns are accessible on distributed hypetable if the optimization is disabled. So I don't think it should be stated in the suggested way.

Comment (pmwkaa) [reply tsl/src/fdw/data_node_scan_exec.c]: Updated the message to be more specific

**Review by k-rus — APPROVED** [review]

**Pull request merged**
